js/skin.js

```diff
@@ -85,30 +86,36 @@ const Skin = createReactClass({
   },
 
   updatePlayhead: function(newPlayhead, newDuration, newBuffered, adPlayhead) {
-    this.setState({
-      currentPlayhead: Utils.ensureNumber(newPlayhead, this.state.currentPlayhead),
-      duration: Utils.ensureNumber(newDuration, this.state.duration),
-      buffered: Utils.ensureNumber(newBuffered, this.state.buffered),
-      currentAdPlayhead: Utils.ensureNumber(adPlayhead, this.state.currentAdPlayhead)
+    let promise = new Promise((resolve) => {
+      this.setState(function(prevState) {
+        const duration = Utils.ensureNumber(newDuration, prevState.duration);
+        const totalTime = this.getTotalTime(duration);
+        const currentPlayhead = Utils.ensureNumber(newPlayhead, prevState.currentPlayhead);
+        const buffered = Utils.ensureNumber(newBuffered, prevState.buffered);
+        const currentAdPlayhead = Utils.ensureNumber(adPlayhead, prevState.currentAdPlayhead);
+        return {
+          currentPlayhead,
+          duration,
+          buffered,
+          currentAdPlayhead,
+          totalTime
+        }
+      }, function() {
```

Comment:
103 - 105 lines could be just `}, resolve});`. No anonymous function wrapper needed here

Reply:
\+